lib/rules/array.js

```diff
@@ -175,12 +175,29 @@ class Length extends Rule {
 }
 
 class Unique extends Rule {
+    compile(params) {
+        const comp = params.args[0];
+        if (typeof comp === 'string') {
+            this._key = comp;
+        } else if (comp instanceof Function) {
+            this._compFn = comp;
+        }
+    }
+
     validate(params, callback) {
         const v = params.value;
         const l = v.length;
         for (let i = 0; i < l; i++) {
             for (let k = i + 1; k < l; k++) {
-                if (eq(v[i], v[k])) {
+                let isEqual = false;
+                if (this._compFn) {
+                    isEqual = this._compFn(v[i], v[k]);
+                } else if (this._key) {
+                    isEqual = eq(get(v[i], this._key), get(v[k], this._key));
+                } else {
+                    isEqual = eq(v[i], v[k]);
```

Comment:
Have a `_comp` property with the default being the `eq` function to simplify logic 😄